main/ServiceManager/ServiceManager.cpp

```diff
@@ -110,16 +161,24 @@ void ServiceMangerInit()
 void ServiceMangerTask(void *pvParameter)
 {
     ServiceMangerInit();
-    char pcTaskList[TASK_LIST_BUFFER_SIZE];
-#ifndef DONE_COMPONENT_MATTER
-    RunMQTTAndTestJson();
+#ifdef CONFIG_DONE_APLLICAION_COFFEE_MAKER
+    bool WifiFlag = CheckWifiStatus();
+    if (WifiFlag)
+    {
+        bool InternetFlag = CheckInternetConnection();
+        if (InternetFlag)
+        {
+            CoffeeMakerApplication(&MQTTDataFromBrokerQueue, &MQTTConnectedSemaphore, &MQTTErrorOrDisconnectSemaphore);
+        }
+    }
```

Comment:
should be like this
```
void ServiceMangerTask(void *pvParameter)
{
    ServiceMangerInit();
    RunMQTTAndTestJson();
    if (IsWifiConnected() == false)
    {
          \\raise an error
          return;
    }
    if (IsInternetConnected() == fasle)
    {
         \\raise an error
         return;
    }

    CoffeeMakerApplication(&MQTTDataFromBrokerQueue, 
                             &MQTTConnectedSemaphore, 
                             &MQTTErrorOrDisconnectSemaphore);
}

Reply:
I fixed it, (I don't need to raise the error, because functions handle errors.)

Reply:
And we can not return from Task.

Reply:
I removed the internet checker function because it was not handling different scenarios.